src/main/java/com/browserstack/automate/ci/jenkins/local/JenkinsBrowserStackLocal.java

```diff
@@ -51,7 +51,7 @@ public JenkinsBrowserStackLocal(String accesskey, LocalConfig localConfig, Strin
             arguments.add(args[i]);
         }
 
-        localIdentifier = UUID.randomUUID().toString().replaceAll("\\-", "") + "-" + buildTag.replaceAll("\\s","").replaceAll("#", "_");
+        localIdentifier = (UUID.randomUUID().toString() + "-" + buildTag).replaceAll("[^a-bA-Z0-9_\\-\\.]", "_");
```

Comment:
documentation says that randomUUID provides numbers so do you need to apply replaceAll also for this string?

https://docs.oracle.com/javase/7/docs/api/java/util/UUID.html#randomUUID()

Reply:
No, not really. The old code did a `replaceAll` on the UUID so I treated the resulting identifier as one but it could equally just be applied to the `buildTag`.